modules/sdk/project-templates/project-templates/src/test/java/com/liferay/project/templates/BaseProjectTemplatesTestCase.java

```diff
@@ -517,13 +571,175 @@ public default File buildWorkspace(
 
 		String name = "test-workspace";
 
-		File destinationDir = temporaryFolder.newFolder("workspace" + name);
+		File destinationDir = temporaryFolder.newFolder(
+			"gradleWorkspace" + name);
 
 		return buildTemplateWithGradle(
 			destinationDir, WorkspaceUtil.WORKSPACE, name, "--liferay-version",
 			liferayVersion);
 	}
 
+	public default File buildWorkspaceWithTPEnabled(
+			TemporaryFolder temporaryFolder, String buildType, String name,
+			String liferayVersion, MavenExecutor mavenExecutor)
+		throws Exception {
+
+		File workspaceDir;
+
+		if (buildType.equals("gradle")) {
+			workspaceDir = buildWorkspace(temporaryFolder, liferayVersion);
+
+			enableTargetPlatformInWorkspace(workspaceDir, liferayVersion);
+		}
+		else {
+			File destinationDir = temporaryFolder.newFolder("mavenWorkspace");
+			String groupId = "com.test";
+
+			workspaceDir = buildTemplateWithMaven(
+				destinationDir, destinationDir, "workspace", name, groupId,
+				mavenExecutor, "-DliferayVersion=" + liferayVersion,
+				"-Dpackage=com.test");
+		}
+
+		return workspaceDir;
+	}
+
+	public default void configureExecutePackageManagerTask(File projectDir)
+		throws Exception {
+
+		File buildGradleFile = new File(projectDir, "build.gradle");
+
+		StringBuilder sb = new StringBuilder();
+
+		String lineSeparator = System.lineSeparator();
+
+		sb.append(lineSeparator);
+
+		sb.append("buildscript {");
+		sb.append(lineSeparator);
+
+		sb.append("\tdependencies {");
+		sb.append(lineSeparator);
+
+		sb.append(
+			"\t\tclasspath group: \"com.liferay\", name: \"com.liferay." +
+				"gradle.plugins\", version: \"5.0.13\"");
+		sb.append(lineSeparator);
+
+		sb.append("\t}");
+		sb.append(lineSeparator);
+
+		sb.append("\trepositories {");
+		sb.append(lineSeparator);
+
+		sb.append("\t\tmaven {");
+		sb.append(lineSeparator);
+
+		sb.append(
+			"\t\t\turl \"https://repository-cdn.liferay.com/nexus/content" +
+				"/groups/public\"");
+		sb.append(lineSeparator);
+
+		sb.append("\t\t}");
+		sb.append(lineSeparator);
+
+		sb.append("\t}");
+		sb.append(lineSeparator);
+
+		sb.append("}");
+		sb.append(lineSeparator);
+
+		sb.append(
+			"import com.liferay.gradle.plugins.node.tasks." +
+				"ExecutePackageManagerTask");
+		sb.append(lineSeparator);
+
+		sb.append("tasks.withType(ExecutePackageManagerTask) {");
```

Comment:
This could be simplified using `withId` and then you don't need the import and classpath stuff.